core/src/main/java/org/kiwix/kiwixmobile/core/history/HistoryPresenter.kt

```diff
@@ -0,0 +1,75 @@
+/*
+ * Kiwix Android
+ * Copyright (c) 2019 Kiwix <android.kiwix.org>
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program. If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+package org.kiwix.kiwixmobile.core.history
+
+import android.util.Log
+import io.reactivex.Observable
+import io.reactivex.Scheduler
+import io.reactivex.disposables.Disposable
+import org.kiwix.kiwixmobile.core.base.BasePresenter
+import org.kiwix.kiwixmobile.core.data.DataSource
+import org.kiwix.kiwixmobile.core.di.qualifiers.Computation
+import org.kiwix.kiwixmobile.core.di.qualifiers.MainThread
+import org.kiwix.kiwixmobile.core.history.HistoryContract.Presenter
+import org.kiwix.kiwixmobile.core.history.HistoryContract.View
+import org.kiwix.kiwixmobile.core.history.HistoryListItem.HistoryItem
+import java.util.Locale
+import javax.inject.Inject
+
+internal class HistoryPresenter @Inject constructor(
+  private val dataSource: DataSource,
+  @param:MainThread private val mainThread: Scheduler,
+  @param:Computation private val computation: Scheduler
+) : BasePresenter<View>(), Presenter {
+
+  private var disposable: Disposable? = null
+
+  override fun loadHistory(showHistoryCurrentBook: Boolean) {
+    val d = dataSource.getDateCategorizedHistory(showHistoryCurrentBook).subscribe(
```

Comment:
we can get rid of this variable, also single letter variable names are the devil

Reply:
disposable is a var and it complains that it could have changed in between being set to the subscription so I used an immutable so that I don't have to use !!.

Reply:
we could avoid this with ..... oooh `let` or `also`.... let probably makes most sense. So a `.let` after `.subscribe` where yo handle assignment and adding to compositeDisposable

Reply:
Much cleaner! I really have a hard time getting used to kotlin-style handling of nulls..

Reply:
This is an annoying situation because the compiler just can't verify that it is not null even though it is clearly not null (technically it could somehow be modified before you executed your next instruction), Let is sometimes not the best way of null checking and sometimes just null checking is better, [dan lew has a great article on it](https://blog.danlew.net/2019/02/12/when-to-use-kotlins-stdlib-functions/)

Reply:
ok! thanks.